app/src/main/java/org/oppia/android/app/onboarding/OnboardingFragmentPresenter.kt

```diff
@@ -36,13 +84,197 @@ class OnboardingFragmentPresenter @Inject constructor(
+    translationController.getSystemLanguageLocale().toLiveData().observe(
+      fragment,
+      { result ->
+        onboardingAppLanguageViewModel.setSelectedLanguageDisplayName(
+          appLanguageResourceHandler.computeLocalizedDisplayName(
+            processSystemLanguageResult(result)
+          )
+        )
+      }
+    )
+  }
+
+  private fun processSystemLanguageResult(
+    result: AsyncResult<OppiaLocale.DisplayLocale>
+  ): OppiaLanguage {
+    return when (result) {
+      is AsyncResult.Success -> {
+        result.value.getCurrentLanguage()
+      }
+      is AsyncResult.Failure -> {
+        oppiaLogger.e(
+          "OnboardingFragment",
+          "Failed to retrieve system language locale.",
+          result.error
+        )
+        OppiaLanguage.ENGLISH
+      }
+      is AsyncResult.Pending -> OppiaLanguage.ENGLISH
+    }
+  }
+
+  private fun getSupportedLanguages() {
+    translationController.getSupportedAppLanguages().toLiveData().observe(
+      fragment,
+      { result ->
+        when (result) {
+          is AsyncResult.Success -> {
+            val supportedLanguages = mutableListOf<String>()
+            result.value.map {
+              supportedLanguages.add(appLanguageResourceHandler.computeLocalizedDisplayName(it))
+              onboardingAppLanguageViewModel.setSupportedAppLanguages(supportedLanguages)
+            }
+          }
+          is AsyncResult.Failure -> {
+            oppiaLogger.e(
+              "OnboardingFragment",
+              "Failed to retrieve supported language list.",
+              result.error
+            )
+          }
+          is AsyncResult.Pending -> {}
+        }
+      }
+    )
+  }
+
+  private fun subscribeToGetProfileList() {
+    existingProfiles.observe(
+      fragment,
+      { profilesList ->
+        if (!profilesList.isNullOrEmpty()) {
+          retrieveProfileId(profilesList)
+        } else {
+          createDefaultProfile()
+        }
+      }
+    )
+  }
+
+  private val existingProfiles: LiveData<List<Profile>> by lazy {
+    Transformations.map(
+      profileManagementController.getProfiles().toLiveData(),
+      ::processGetProfilesResult
+    )
+  }
+
+  private fun processGetProfilesResult(profilesResult: AsyncResult<List<Profile>>): List<Profile> {
+    val profileList = when (profilesResult) {
+      is AsyncResult.Failure -> {
+        oppiaLogger.e(
+          " OnboardingFragment", "Failed to retrieve the list of profiles", profilesResult.error
+        )
+        emptyList()
+      }
+      is AsyncResult.Pending -> emptyList()
+      is AsyncResult.Success -> profilesResult.value
+    }
+
+    return profileList
+  }
+
+  private fun createDefaultProfile() {
+    profileManagementController.addProfile(
+      name = "Admin", // TODO(#4938): Refactor to empty name once proper admin profile creation flow
+      // is implemented.
+      pin = "",
+      avatarImagePath = null,
+      allowDownloadAccess = allowDownloadAccess,
```

Comment:
Can be true always, I think, since the feature is off (and when we enable it in the future there'd be no way to retroactively turn this back on, I think, without a required profile migration).